acme/api/order.go

```diff
@@ -97,6 +103,40 @@ func (h *Handler) NewOrder(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	// TODO(hs): gather all errors, so that we can build one response with subproblems; include the nor.Validate()
+	// error here too, like in example?
+
+	eak, err := h.db.GetExternalAccountKeyByAccountID(ctx, prov.GetID(), acc.ID)
+	if err != nil {
+		render.Error(w, acme.WrapErrorISE(err, "error retrieving external account binding key"))
+		return
+	}
```

Comment:
Checking if EAB is required is a nice suggestion, indeed. Saves a query to the DB. Fixed in fb81407d6f294a229fd833c52fcbd91bf85a8590.

I tend to disagree on returning an `ErrNotFound` vs. nil. I think the application code is responsible for the case in which a query returned no object if it didn't exist in the database. "Leaking" that error from the data storage layer IMO makes it more work to handle within the application. It's possible for an object to not exist when queried in some way; that's generally not an error case, IMO. The underlying data service/repository may have a need for that `ErrNotFound` (e.g. if the layer below that returns something like that), but it can be translated into a nil object returned to the application layer.